akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -223,8 +223,10 @@ import akka.util.OptionVal
       log.error(ex, logMessage)
       if (isTypedActor)
         classic.SupervisorStrategy.Stop
-      else
-        classic.SupervisorStrategy.Restart
+      else {
+        // ActorInitializationException => Stop in defaultDecider
+        classic.SupervisorStrategy.defaultDecider.applyOrElse(ex, (_: Throwable) => classic.SupervisorStrategy.Restart)
```

Comment:
Should we cache the function, or even the entire decider?

Reply:
ok: https://github.com/akka/akka/pull/27991/commits/fceca136b98e13aa58d6667a9d9cb415bf5c05c4